lib/core/jtrace.c

```diff
@@ -550,6 +742,275 @@ j_trace_enter(gchar const* name, gchar const* format, ...)
 		G_UNLOCK(j_trace_echo);
 	}
 
+	if (j_trace_flags & J_TRACE_ACCESS)
+	{
+		if (strcmp(name, "ping") == 0)
+		{
+			va_list args;
+			va_start(args, format);
+			g_free(trace_thread->client.program_name);
+			trace_thread->client.program_name = strdup(va_arg(args, const char*));
+			trace_thread->client.uid = va_arg(args, guint32);
+			va_end(args);
+		}
+		else if (strncmp(name, "backend_", 8) == 0)
+		{
+			int type = 0;
+			if (trace_thread->access.db.type == NULL) /// \TODO more precise checking?
```

Comment:
What do you mean by precise checking?

Reply:
This part is for loading the `type` and `config_path` of each backend, which cannot be done in the `init `since the trace will be initialized before the backends.

Instead of checking `db.type`, it could be checked more of null, but this would mean an invalid state .... so think the test is enough, and will remove the todo